src/SFA.DAS.AANHub.Application/Members/Queries/GetMemberActivities/GetMemberActivitiesQueryHandler.cs

```diff
@@ -0,0 +1,63 @@
+﻿using MediatR;
+using SFA.DAS.AANHub.Application.Mediatr.Responses;
+using SFA.DAS.AANHub.Application.Models;
+using SFA.DAS.AANHub.Domain.Interfaces.Repositories;
+using static SFA.DAS.AANHub.Application.Constants;
+
+namespace SFA.DAS.AANHub.Application.Members.Queries.GetMemberActivities;
+public class GetMemberActivitiesQueryHandler : IRequestHandler<GetMemberActivitiesQuery, ValidatedResponse<GetMemberActivitiesResult>>
+{
+    private readonly IAttendancesReadRepository _attendancesReadRepository;
+    private readonly IAuditReadRepository _auditReadRepository;
+
+    public GetMemberActivitiesQueryHandler(IAttendancesReadRepository attendancesReadRepository, IAuditReadRepository auditReadRepository)
+    {
+        _attendancesReadRepository = attendancesReadRepository;
+        _auditReadRepository = auditReadRepository;
+    }
+
+    public async Task<ValidatedResponse<GetMemberActivitiesResult>> Handle(GetMemberActivitiesQuery request, CancellationToken cancellationToken)
+    {
+        DateTime rangeStartDate = DateTime.UtcNow.AddMonths(-1 * RangeDuration.EventsRangePeriod);
+        DateTime rangeEndDate = DateTime.UtcNow.AddMonths(RangeDuration.EventsRangePeriod);
+        DateTime currentDate = DateTime.UtcNow;
+
+        var memberActivities = await _attendancesReadRepository.GetAttendances(request.MemberId, rangeStartDate, rangeEndDate, cancellationToken);
```

Comment:
There should be test to assert on the date range that is calculated here.

Reply:
Done